szamlazz/models.py

```diff
@@ -147,8 +147,8 @@ def __init__(self,
         self.xml_namespace = xml_namespace
         self.__response = response
         self.__action_success: bool = False
-        content_type = response.headers.get("Content-Type")
-        if content_type == "application/octet-stream":
+        content_type = response.headers.get("Content-Type", "")
+        if "application/octet-stream" in content_type:
```

Comment:
When comparing strings in Python we want to compare values and for that the proper way is using the equality (`==`) operator. Using the `in` keyword opens up the possibility to matching with values that may (by error) contain something else besides `application/octet-stream`. Matching excatly means gives better realiability in my opinion.

Reply:
Exact equality check is incorrect in this case, because this header may include [other info](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type) (such as character encoding, which Számlázz now sends). The client is currently broken because of this.

Reply:
you're right, this was an oversight on my end. i'm merging this PR, however, i'm still uncertain about setting a default value in line 150 so i'm overriding that and if you still find that necessary then pls discuss it further. thanks!

Reply:
That just simplifies the check, it doesn't really matter if content type is `None` or empty string, but sure, your repo, your preferences. Thanks!

Reply:
Did you actually observe a `TypeError` or is it just an assumption that it may happen?
I'm asking because according to the [RFC 9110 text](https://www.rfc-editor.org/rfc/rfc9110.html#name-content-type), the `Content-Type` header should be present in all cases except when the sender itself doesn't know the type which is impossible in our case.

Adding a default value when the CT header is missing would actually mask a server programming error on the client side and I'm not feeling comfortable with that.

In case you actually observed a `TypeError` raise then please provide me some details to reproduce it, and so, I can fill in a bug ticket at Szamlazz.hu because that's inappropriate server behaviour.

Many thanks for your contributions so far!

Reply:
I didn't want to get into an argument on this topic, but okay. I'll keep it short. The duty of a client is not to validate whether the service obeys certain standards, but to work safely in as many scenarios as possible. In this case, it is not guaranteed. If you used a static code analyzer, it would immediately flag this as an error.

I did not observe the error, but I also did not observe the one I fixed in my merge request until Szamlazz changed something that triggered the bug in the client. Sidenote: I'm still using a patched version of the lib for the reasons above.

Reply:
@volfpeter Sorry about my delayed reply, I got occupied with other things then forgot about this discussion.
I've reconsidered my previous opinion and I've added this change to the `.get()` call. Thanks for your contribution!